File: FirstApplication/app/src/main/java/com/epn/fistyapplication/MainActivity.java
package com.epn.fistyapplication;

import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    TextView resultadoNombre;
    EditText nombreEditText;
    public void buttonPress(View view){
        Log.i("Info", "Boton Presionado");
        nombreEditText = findViewById(R.id.editNombre);
        String nombreString = nombreEditText.getText().toString();
        resultadoNombre.setText(nombreString);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        resultadoNombre = findViewById(R.id.tvResultadoNombre);
        resultadoNombre.setText("NOMBRE AQUÍ");

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }
}

File: ConsumoApiRest/app/src/main/java/com/epn/mishelramirez/consumoapirest/MainActivity.kt
package com.epn.mishelramirez.consumoapirest

import android.os.Bundle
import android.util.Log
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.epn.mishelramirez.consumoapirest.implementacion.ApiClientes
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import kotlinx.coroutines.Dispatchers

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        //lifecycleScope.launch(Dispatchers.IO) Esta uso el profe
        CoroutineScope(Dispatchers.IO).launch {
            try {
                val clientes= ApiClientes.api.getClientes()
                withContext(Dispatchers.Main){
                    clientes.forEach {
                        Log.d("API", it.Id.toString())
                        Log.d("API -- Nombre", it.Nombre.toString())
                    }
                }
            }catch (e: Exception){
                Log.e("Error",e.message!!);
            }

        }
    }
}